packages/tpetra/core/ext/TpetraExt_MatrixMatrix_Cuda.hpp

```diff
@@ -170,7 +170,20 @@ void KernelWrappers<Scalar,LocalOrdinal,GlobalOrdinal,Tpetra::KokkosCompat::Kokk
   KokkosSparse::SPGEMMAlgorithm alg_enum = KokkosSparse::StringToSPGEMMAlgorithm(myalg);
 
   // Merge the B and Bimport matrices
-  const KCRS Bmerged = Tpetra::MMdetails::merge_matrices(Aview,Bview,Acol2Brow,Acol2Irow,Bcol2Ccol,Icol2Ccol,C.getColMap()->getLocalNumElements());
+  KCRS Bmerged = Tpetra::MMdetails::merge_matrices(Aview,Bview,Acol2Brow,Acol2Irow,Bcol2Ccol,Icol2Ccol,C.getColMap()->getLocalNumElements());
+
+
+  // if Kokkos Kernels is going to use the cuSparse TPL for SpGEMM, this matrix
+  // needs to be sorted
+  // Try to mirror the Kokkos Kernels internal SpGEMM TPL use logic
+  // inspired by https://github.com/trilinos/Trilinos/pull/11709
```

Comment:
I've followed this link and taken a look at PR #11709. However, it seems it's a closed PR, and it doesn't seem to bring the logic with the check of the cuda version. 

So these are a few thoughts:
- It may be worth double checking that is it indeed #11709 that you intend to reference. Or perhaps there is an additional code snippet with the internal SpGEMM TPL use logic?
- Depending on the intent of the check of the cuda version, it may be useful to see whether it could become a check of the cusparse version, with perhaps a short comment about what changed in the excluded versions. 
- It looks like in PR #11709, the sorts were protected with checks of whether or not the matrix graph was already sorted. Would it be useful to use such checks here too?

Reply:
I think you're right about checking whether the CrsGraph is sorted, I will add that.

The point of the other logic is just to mirror what Kokkos Kernels does, since that determines whether the TPL is used. That was introduced in https://github.com/kokkos/kokkos-kernels/pull/2008 and subsequently modified, I can note that in the comment as well

Reply:
OK. Got it! :)